File: TrackIt.Services/Database/Meal.cs
namespace TrackIt.Services.Database;

public partial class Meal
{
	public int MealId { get; set; }

	public double? Fat { get; set; }

	public double? Calories { get; set; }

	public double? Carbs { get; set; }

	public double? Protein { get; set; }

	public string? Name { get; set; }

	public string? Description { get; set; }

	public byte[]? Image { get; set; }

	public virtual ICollection<MealsIngredient> MealsIngredients { get; set; } = new List<MealsIngredient>();

	public virtual ICollection<TagsMeal> TagsMeals { get; set; } = new List<TagsMeal>();

	public virtual ICollection<UsersMeal> UsersMeals { get; set; } = new List<UsersMeal>();
}

File: UI/trackit_admin/lib/models/Meal/meal.dart
import 'package:json_annotation/json_annotation.dart';

import '../MealIngredient/meal_ingredient.dart';
part 'meal.g.dart';

@JsonSerializable()
class Meal {
  int? mealId;
  double? fat;
  double? calories;
  double? carbs;
  double? protein;
  String? name;
  String? description;
  String? image;
  List<MealIngredient>? mealsIngredients;

  Meal(this.mealId, this.fat, this.calories, this.carbs, this.protein,
      this.name, this.description, this.image, this.mealsIngredients);

  factory Meal.fromJson(Map<String, dynamic> json) => _$MealFromJson(json);

  Map<String, dynamic> toJson() => _$MealToJson(this);
}
